backend/controllers/LocationsController.ts

```diff
@@ -1,40 +1,47 @@
-import { Response, Request } from "express";
-import Location from "../models/Locations";
+import { Response, Request } from 'express';
+import { TupleTypeReference } from 'typescript';
+import Location from '../models/Locations';
 
 const LocationsController = {
   async getLocationsByZoom(req: Request, res: Response) {
     try {
-      const center: [number, number] = [
-          JSON.parse(req.body.center).lat,
-          JSON.parse(req.body.center).lng,
-        ],
-        bounds = JSON.parse(req.body.bounds);
-
+      const center = JSON.parse(req.query.center as string);
+      const bounds = JSON.parse(req.query.bounds as string);
+      const name = req.query.name as string;
+      const filters = JSON.parse(req.query.filters as any)
       const height = +(bounds._northEast.lat - bounds._southWest.lat);
       const width = +(bounds._northEast.lng - bounds._southWest.lng);
-
       let locations = (
         await Location.find({
-          "coordinates.0": {
-            $gt: center[0] - height,
-            $lt: center[0] + height,
+          'coordinates.0': {
+            $gt: center.lat - height,
+            $lt: center.lat + height,
           },
-          "coordinates.1": {
-            $gt: center[1] - width,
-            $lt: center[1] + width,
+          'coordinates.1': {
+            $gt: center.lng - width,
+            $lt: center.lng + width,
           },
         })
       ).map((l) => ({
         _id: l._id,
         coordinates: l.coordinates,
+        name:l.locationName,
+        filters:l.filters
       }));
-
-      locations = locations.slice(
-        0,
-        locations.length < 50 ? locations.length : 50
-      );
-
-      return res.json({ locations });
+      if(name){
+        locations = locations.filter((l)=>{
+          return l.name.toLocaleLowerCase().startsWith(name)
+        })
```

Comment:
A search implementation is much harder than that.
[one of implementations](https://www.digitalocean.com/community/tutorials/how-to-perform-full-text-search-in-mongodb)
Maybe, it should be a separate task though.

Reply:
yeap, we should create a task to change the filtering strategy

Reply:
at next sprint - you should resolve this conversation